paimon-core/src/main/java/org/apache/paimon/mergetree/compact/ChangelogMergeTreeRewriter.java

```diff
@@ -170,6 +175,7 @@ public CompactResult upgrade(int outputLevel, DataFileMeta file) throws Exceptio
                     outputLevel,
                     Collections.singletonList(
                             Collections.singletonList(SortedRun.fromSingle(file))),
+                    false,
```

Comment:
No need, because we don’t need to rewrite when upgrading, even if rewrite is required, this drop delete must be false too (can take a look at the conditions for drop delete and rewrite in upgrade).

Reply:
Thanks for explanation. Looks reasonable, if rewrite is required, the file level always in level 0.